src/main/java/org/joda/time/tz/ZoneInfoCompiler.java

```diff
@@ -168,7 +168,7 @@ static void writeZoneInfoMap(DataOutputStream dout, Map<String, DateTimeZone> zi
                 Short index = Short.valueOf(count);
                 idToIndex.put(id, index);
                 indexToId.put(index, id);
-                if (++count == 0) {
+                if (++count == Integer.MAX_VALUE) {
```

Comment:
This seems like it might be a bug? How can `count` (which is a `short`), ever approach `Integer.MAX_VALUE`?

Should this have been `Short.MAX_VALUE`? Ditto on L180